app/modules/enduser_setup.c

```diff
@@ -626,6 +626,52 @@ static void do_station_cfg (task_param_t param, uint8_t prio)
 }
 
 
+/**
+ *
+ *
+ *
+ *
+ */
+static int enduser_setup_write_file_with_extra_configuration_data(
+char * configuration_string, int configuration_length
+)
+{
+  int p_file = 0;
+  lua_State *L = NULL;
+	
+  ENDUSER_SETUP_DEBUG("enduser: opening enduser_custom_parameters.json for write");
+
+  // setup the file output
+  p_file = vfs_open("enduser_custom_parameters.json", "w");
+  if (p_file == 0)
+  {
+    ENDUSER_SETUP_DEBUG("Can't write to file!");
+    return 1;
+  }
+  
+  /* Not certain what this does */
```

Comment:
This gets you the correct Lua context to work on. Note that this `lua_State` stack might already be in use, so care must be taken to ensure that the stack is left in the same condition as when the function was entered. Weird Stuff(tm) happens otherwise, of the variety that's hard to track down.